full-service/src/unsigned_tx.rs

```diff
@@ -53,13 +59,29 @@ impl UnsignedTx {
         fog_resolver: FullServiceFogResolver,
     ) -> Result<TxProposal, WalletTransactionBuilderError> {
         let mut rng = rand::thread_rng();
+
         // Create transaction builder.
-        let mut memo_builder = RTHMemoBuilder::default();
-        memo_builder.set_sender_credential(SenderMemoCredential::from(account_key));
-        memo_builder.enable_destination_memo();
         let fee = Amount::new(self.fee, TokenId::from(self.fee_token_id));
-        let mut transaction_builder =
-            TransactionBuilder::new(self.block_version, fee, fog_resolver, memo_builder)?;
+
+        let mut transaction_builder = match self.memo {
+            TransactionMemo::RTH => {
```

Comment:
Looks like this is duplicated from the signed tx flow.
Can this become a method on the `TransactionMemo` type? e.g. `fn get_boxed_memo_builder(&self, account_key: &AccountKey) -> Box<dyn ...>`?

Reply:
Yeah, I have a separate PR coming to refactor the transaction building process into build and sign which everything will work off of